### ISSUE TITLE
allowable HTTP status code

### ISSUE THREAD

**Issue by toyoshim** [WICG/webpackage#478]
Currently, file formats just expect the status code should be exactly 3 ASCII decimal digits.

But for SignedExchanges, the "Loading Signed Exchanges" spec allows only 200 for the status code (https://wicg.github.io/webpackage/loading.html#parse-cbor-headers step 2).

Probably, BundledExchanges also needs a similar restriction, but probably we want range requests and redirects work (206, 307, 308).

For Chrome implementation, we will start from just allowing only 200 for the initial experimental implementation, but will support others if we want.

crbug.com/990733 tracks the Chrome side implementation.

**Comment by toyoshim** [comment]
cc: @irori @kinu @jyasskin

**Comment by toyoshim** [comment]
Here is the discussion for SignedExchange case; https://github.com/WICG/webpackage/pull/326

**Comment by jyasskin** [comment]
I think it probably makes sense to allow redirects in the bundle, but given the complexity of partial responses, I'd be inclined to keep them out until we find a more concrete use case. A bundle could still serve a partial request when it contains a complete response, although in cases where that's important, we probably also want a more seekable encoding than mi-sha256.

For now, +1 on having the Chrome implementation only handle 200s.

**Comment by kinu** [comment]
I agree that it could be useful to support range requests / 206. (We were looking into a use case where we want to support video resources in a bundle)  While- if we want to support it with mi-sha256 UA still probably needs to read everything in-memory or somewhere, so you're right that we might want a more seekable encoding :thinking:

**Comment by robbiemc** [comment]
[Isolated Web Apps](https://github.com/WICG/isolated-web-apps) use Web Bundles as their packaging format, and we'd like to support redirects in the bundles. We currently only allow 200s to follow the spec, but would like to at least allow 301/8 (maybe others?).

Is there still interest in expanding the allowed response codes? For our use case I don't see a reason to restrict the responses at all, but I imagine for other use cases it would make sense. Would 301 and 308 make sense more broadly?
